File: scratch/link_test.py
# Test link generation for Arbitrum pools

def generate_links(network, pool_addr):
    network_lower = network.lower()
    # Uniswap mapping
    if 'eth' in network_lower:
        uni_network = 'ethereum'
    elif 'bnb' in network_lower:
        uni_network = 'bsc'
    elif 'arbitrum' in network_lower:
        uni_network = 'arbitrum'
    elif 'optimism' in network_lower:
        uni_network = 'optimism'
    elif 'polygon' in network_lower:
        uni_network = 'polygon'
    elif 'base' in network_lower:
        uni_network = 'base'
    else:
        uni_network = 'ethereum'
    uniswap_link = f"https://app.uniswap.org/explore/pools/{uni_network}/{pool_addr}"

    # PancakeSwap mapping
    if 'eth' in network_lower:
        p_chain = 'eth'
    elif 'arbitrum' in network_lower:
        p_chain = 'arbitrum'
    elif 'base' in network_lower:
        p_chain = 'base'
    else:
        p_chain = 'bsc'
    pancake_link = f"https://pancakeswap.finance/info/v3/pairs/{pool_addr}?chain={p_chain}"
    return uniswap_link, pancake_link

addr = "0x67D3E181E6dcC47f977c3A4b33Ac65454b87b997"
uni, pan = generate_links('Arbitrum', addr)
print('Uniswap link:', uni)
print('Pancake link:', pan)

File: scratch/link_test.js
// Simple test to verify link generation for Arbitrum pools
function generateLinks(network, poolAddr) {
  const networkLower = network.toLowerCase();
  // Uniswap mapping
  let uniNetwork = 'ethereum';
  if (networkLower.includes('base')) uniNetwork = 'base';
  else if (networkLower.includes('eth')) uniNetwork = 'ethereum';
  else if (networkLower.includes('bnb') || networkLower.includes('bsc')) uniNetwork = 'bnb';
  else if (networkLower.includes('arbitrum')) uniNetwork = 'arbitrum';
  else if (networkLower.includes('optimism')) uniNetwork = 'optimism';
  else if (networkLower.includes('polygon')) uniNetwork = 'polygon';

  const uniswapLink = `https://app.uniswap.org/explore/pools/${uniNetwork}/${poolAddr}`;

  // PancakeSwap mapping
  let pChain = 'bsc';
  if (networkLower.includes('base')) pChain = 'base';
  else if (networkLower.includes('eth')) pChain = 'eth';
  else if (networkLower.includes('arbitrum')) pChain = 'arb';
  const pancakeLink = `https://pancakeswap.finance/info/v3/pairs/${poolAddr}?chain=${pChain}`;

  return { uniswapLink, pancakeLink };
}

const addr = '0x67D3E181E6dcC47f977c3A4b33Ac65454b87b997';
const links = generateLinks('Arbitrum', addr);
console.log('Uniswap link:', links.uniswapLink);
console.log('Pancake link:', links.pancakeLink);
